README.md
# pyLMGF
Lords Mobile Guild Fest Bot


history.txt
2020 07 19 02 39 55 43 0 Khemmis
2020 07 19 02 58 52 157 0 Khemmis
2020 07 19 02 58 58 184 0 Khemmis
2020 07 19 02 59 03 49 0 Khemmis
2020 07 19 02 59 08 173 0 Khemmis
2020 07 19 02 59 12 143 0 Khemmis
2020 07 19 02 59 15 101 0 Khemmis
2020 07 19 02 59 19 161 0 Khemmis
2020 07 19 02 59 25 23 0 Khemmis
2020 07 19 02 59 28 140 0 Khemmis
2020 07 19 02 59 33 166 0 Khemmis
2020 07 19 02 59 37 28 0 Khemmis
2020 07 19 02 59 41 147 0 Khemmis
2020 07 19 02 59 46 185 0 Khemmis
2020 07 19 02 59 55 27 0 Khemmis
2020 07 19 03 13 10 146 0 XDodie
2020 07 19 03 13 11 146 0 XDodie
2020 07 19 03 38 58 240 0 XDodie
2020 07 19 03 39 01 241 0 XDodie
2020 07 19 03 39 05 215 0 XDodie
2020 07 19 03 39 09 247 0 XDodie
2020 07 19 03 39 14 242 0 XDodie
2020 07 19 03 39 17 207 0 XDodie
2020 07 19 03 40 07 166 1 XDodie
2020 08 03 15 22 59 252 0 XDodie
2020 08 03 15 32 30 252 0 XDodie
2020 08 03 15 32 31 203 0 XDodie
2020 08 03 15 32 31 146 0 XDodie
2020 08 03 15 32 31 283 0 XDodie
2020 08 03 15 32 31 247 0 XDodie
2020 08 03 15 32 31 241 0 XDodie
2020 08 03 15 32 31 254 0 XDodie
2020 08 03 15 32 31 240 0 XDodie
2020 08 03 15 32 31 241 0 XDodie
2020 08 03 15 32 31 215 0 XDodie
2020 08 03 15 32 32 247 0 XDodie
2020 08 03 15 32 32 166 0 XDodie
2020 08 03 15 32 32 229 0 XDodie
2020 08 03 15 32 32 278 0 XDodie
2020 08 03 15 32 32 215 0 XDodie
2020 08 03 15 32 32 184 0 XDodie
2020 08 03 15 32 32 254 0 XDodie
2020 08 03 15 32 33 268 0 XDodie
2020 08 03 18 15 04 252 0 XDodie
2020 08 03 18 15 04 203 0 XDodie
2020 08 03 18 15 04 146 0 XDodie
2020 08 03 18 15 04 283 0 XDodie
2020 08 03 18 15 04 247 0 XDodie
2020 08 03 18 15 05 241 0 XDodie
2020 08 03 18 15 05 254 0 XDodie
2020 08 03 18 15 05 240 0 XDodie
2020 08 03 18 15 05 241 0 XDodie
2020 08 03 18 15 05 215 0 XDodie
2020 08 03 18 15 05 247 0 XDodie
2020 08 03 18 15 05 241 0 XDodie
2020 08 03 18 15 05 219 0 XDodie
2020 08 03 18 15 06 166 0 XDodie
2020 08 03 18 15 06 229 0 XDodie
2020 08 03 18 15 06 278 0 XDodie
2020 08 03 18 15 06 215 0 XDodie
2020 08 03 18 15 06 184 0 XDodie
2020 08 03 18 15 06 254 0 XDodie
2020 08 03 18 15 06 268 0 XDodie
2020 08 04 01 46 10 220 0 Khemmis
2020 08 04 01 46 12 224 0 Khemmis
2020 08 04 01 46 15 265 0 Khemmis
2020 08 04 01 46 18 24 0 Khemmis
2020 08 04 01 46 20 284 0 Khemmis
2020 08 04 01 46 22 9 0 Khemmis
2020 08 04 01 46 24 289 0 Khemmis
2020 08 04 01 46 26 50 0 Khemmis
2020 08 04 01 46 29 178 0 Khemmis
2020 08 04 01 46 31 17 0 Khemmis
2020 08 04 01 46 33 25 0 Khemmis
2020 08 04 01 46 36 257 0 Khemmis
2020 08 04 01 46 38 64 0 Khemmis
2020 08 04 01 46 40 224 0 Khemmis
2020 08 04 01 46 42 251 0 Khemmis
2020 08 04 02 02 15 128 0 Khemmis
2020 08 04 02 02 18 282 0 Khemmis
2020 08 04 02 02 27 258 0 Khemmis
2020 08 04 02 04 18 223 0 Khemmis
2020 08 04 02 16 23 251 0 Khemmis
2020 08 04 02 16 25 30 0 Khemmis
2020 08 04 02 16 29 236 0 Khemmis
2020 08 04 02 16 32 159 0 Khemmis
2020 08 04 02 16 33 189 0 Khemmis
2020 08 04 02 16 36 126 0 Khemmis
2020 08 04 02 16 40 264 0 Khemmis
2020 08 04 02 16 42 262 0 Khemmis
2020 08 04 02 16 46 199 0 Khemmis
2020 08 04 02 16 48 44 0 Khemmis
2020 08 04 02 16 50 107 0 Khemmis
2020 08 04 02 16 53 145 0 Khemmis
2020 08 04 02 16 55 174 0 Khemmis
2020 08 04 02 19 38 96 0 Khemmis
2020 08 04 02 19 52 178 0 Khemmis
2020 08 04 02 25 03 280 0 Khemmis
2020 08 04 02 36 12 277 0 Khemmis
2020 08 04 02 36 15 199 0 Khemmis
2020 08 04 02 36 19 65 0 Khemmis
2020 08 04 02 47 04 199 0 Khemmis
2020 08 04 02 53 39 77 0 Khemmis
2020 08 04 02 53 40 226 0 Khemmis
2020 08 04 02 53 44 278 0 Khemmis
2020 08 04 02 53 46 65 0 Khemmis
2020 08 04 02 53 49 248 0 Khemmis
2020 08 04 02 53 51 254 0 Khemmis
2020 08 04 02 53 54 172 0 Khemmis
2020 08 04 02 53 57 195 0 Khemmis
2020 08 04 02 53 59 156 0 Khemmis
2020 08 04 02 54 06 123 0 Khemmis
2020 08 04 02 54 08 60 0 Khemmis
2020 08 04 02 54 11 171 0 Khemmis
2020 08 04 02 54 13 242 0 Khemmis
2020 08 04 02 57 26 186 0 Khemmis
2020 08 04 02 57 29 225 0 Khemmis
2020 08 04 03 25 19 173 0 Khemmis
2020 08 04 03 27 56 64 0 Khemmis
2020 08 04 03 32 41 273 0 Khemmis
2020 08 04 03 33 00 223 0 Khemmis
2020 08 04 03 33 02 143 0 Khemmis
2020 08 04 03 33 05 233 0 Khemmis
2020 08 04 03 33 07 92 0 Khemmis
2020 08 04 03 33 11 202 0 Khemmis
2020 08 04 03 33 13 154 0 Khemmis
2020 08 04 03 33 20 27 0 Khemmis
2020 08 04 03 33 22 160 0 Khemmis
2020 08 04 03 33 26 151 0 Khemmis
2020 08 04 03 33 28 276 0 Khemmis
2020 08 04 03 33 36 128 0 Khemmis
2020 08 04 03 33 41 129 0 Khemmis
2020 08 04 03 33 46 268 0 Khemmis
2020 08 04 03 33 49 70 0 Khemmis
2020 08 04 04 04 25 123 0 Khemmis
2020 08 04 04 07 05 237 0 Khemmis
2020 08 04 04 51 23 251 0 Khemmis
2020 08 04 05 24 00 157 0 Khemmis
2020 08 04 05 24 05 130 0 Khemmis
2020 08 04 05 24 08 126 0 Khemmis
2020 08 04 05 24 10 247 0 Khemmis
2020 08 04 05 24 14 288 0 Khemmis
2020 08 04 05 24 17 25 0 Khemmis
2020 08 04 05 24 19 189 0 Khemmis
2020 08 04 05 24 21 252 0 Khemmis
2020 08 04 05 24 23 98 0 Khemmis
2020 08 04 05 24 28 177 0 Khemmis
2020 08 04 05 24 34 12 0 Khemmis
2020 08 04 05 24 36 48 1 Khemmis
2020 09 14 12 33 23 29 0 Khemmis
2020 09 14 12 33 23 77 0 Khemmis
2020 09 14 12 33 24 112 0 Khemmis
2020 09 14 12 33 24 100 0 Khemmis
2020 09 14 12 34 51 180 0 Khemmis
2020 09 14 12 35 19 185 0 Khemmis
2020 09 14 12 35 23 154 0 Khemmis
2020 09 15 22 29 32 167 0 Khemmis
2020 09 15 22 30 33 186 0 Khemmis
2020 09 15 22 36 10 125 0 Khemmis
2020 09 15 22 37 37 101 0 Khemmis
2020 09 15 22 39 43 110 0 Khemmis
2020 09 15 22 40 00 53 0 Khemmis
2020 09 15 22 40 59 89 0 Khemmis
2020 09 15 22 41 23 91 0 Khemmis
2020 09 15 22 41 31 44 0 Khemmis
2020 09 15 23 07 41 164 0 Khemmis
2020 09 15 23 17 26 28 0 Khemmis
2020 09 15 23 19 02 116 0 Khemmis
2020 09 15 23 19 06 22 0 Khemmis
2020 09 15 23 27 25 38 0 Khemmis
2020 09 15 23 27 26 62 0 Khemmis
2020 09 15 23 27 30 165 0 Khemmis
2020 09 15 23 27 34 159 0 Khemmis
2020 09 15 23 27 39 39 0 Khemmis
2020 09 15 23 30 42 148 0 Khemmis
2020 09 15 23 38 16 120 0 Khemmis
2020 09 15 23 41 23 172 0 Khemmis
2020 09 15 23 41 26 76 0 Khemmis
2020 09 15 23 41 29 173 0 Khemmis

hisory_gui.py
import sqlite3
import tkinter as tk
import random
import datetime
from databases import QuestDB
from PIL import ImageTk, Image
db = QuestDB()

if __name__ == '__main__':
    top = tk.Tk
else:
    top = tk.Toplevel

def generate_test():
    names = ['Khemmis', 'THALLES 14', 'Dodie', 'xleo2']
    time_now = int(datetime.datetime.now().timestamp())
    slots = {1:'q',2:'q',3:'q',4:'q',5:'q',6:'q',7:'q',8:'q',9:'q',10:'q',11:'q',12:'q',13:'q',14:'q',15:'q',16:'q',17:'q',18:'q',19:'q',20:'q',21:'q'}

    with open('test.txt', 'a') as f:
        f.write(f'{random.choice(names)}|{time_now}|{",".join([str(random.randint(1, 290)) for _ in range(5)])}:')

    values = []
    for i in range(300):
        slot, identify = random.choice(list(slots.items()))
        if identify == 'q':
            time = 30*60
            slots.update({slot: 't'})
            value = random.randint(1, 290)
        else:
            time = random.randint(1, 30*60)
            slots.update({slot: 'q'})
            value = 30*60
        values.append(f'{time}.{slot}.{identify}.{value}')

    with open('test.txt', 'a') as f:
        f.write('|'.join(values)+'\n')


#generate_test()


class Label(tk.Label):
    def __init__(self, parent, **kw):
        super(Label, self).__init__(parent)
        self.configure(
            bg='light pink',
            font=('Gadugi', 12, 'bold')
        )
        self.config(**kw)


class Slot(tk.Label):
    def __init__(self, parent, number, **kw):
        super(Slot, self).__init__(parent)
        self.config(**kw)
        self.number = number
        self.qid = None
        self.timer = None
        self.sequence = []


class GUI(top):
    def __init__(self, parent=None):
        super(GUI, self).__init__(parent)
        g = ChooseGUI(self)

        self.minsize(500, 500)
        self.images = set([])

        self.slots = {}
        self.name = ''
        self.sequence = []
        self.counter = 1

        self.board_frame = tk.Frame(self, bg='yellow')
        self.board_frame.place(relx=0, rely=0, relwidth=.66, relheight=1)

        self.quest_frame = tk.Frame(self, bg='light cyan')
        self.quest_frame.place(relx=.66, rely=0, relwidth=.34, relheight=.5)

        self.controls_frame = tk.Frame(self, bg='lemon chiffon')
        self.controls_frame.place(relx=.66, rely=.5, relwidth=.34, relheight=.5)

        self.populate_board_frame()
        # self.parse()
        # self.update()

    def populate_board_frame(self):
        for i in range(22):
            slot = Slot(self.board_frame, i, bg='cyan', compound='top')
            slot.place(relx=i % 3 / 3, rely=i // 3 / 7, relwidth=1/3-.01, relheight=1/7-.01)
            self.slots.update({i: slot})
    #
    # def parse(self):
    #     with open('test.txt', 'r') as f:
    #         line = f.readline()
    #
    #     sep = line.find(':')
    #     self.name = line[:sep]
    #     self.sequence = (i for i in line[sep+1:].split('|'))
    #     print(self.sequence)
    #
    # def update(self):
    #     print(self.counter)
    #     self.counter += 1
    #
    #     item = next(self.sequence).strip()
    #     if not item:
    #         return
    #
    #     clock, slot, value = item.split('.')
    #     if value.startswith('q'):
    #         qid = value[1:]
    #         _, text, points, *_, img_index, _ = db.get_quest_by_id(qid)
    #
    #         bg = 'goldenrod'
    #         text = f'+{points}'
    #         self.update_current_quest(qid=qid)
    #     else:
    #         value = int(value[1:])
    #         text = f'{value // 60:02d}:{value % 60:02d}'
    #         bg = 'orange4'
    #         self.update_current_quest(timer=value)
    #
    #     slot = self.slots.get(int(slot))
    #     slot.configure(text=text, bg=bg, fg='white', compound='top', font=('Gadugi', 16, 'bold'))
    #
    #     self.after(100, self.update)
    #
    # def update_current_quest(self, qid=None, timer=None):
    #     if qid is not None:
    #         _, name, points, req, time, *_, quest_img, icon_img = db.get_quest_by_id(qid)
    #         print(name, points, req, time, quest_img, icon_img)
    #         [w.destroy() for w in self.quest_frame.pack_slaves()]
    #
    #         quest_img = ImageTk.PhotoImage(Image.open(f'imgs\\quests\\{quest_img:03d}.png'))
    #         points_img = ImageTk.PhotoImage(Image.open(f'imgs\\icons\\066.png').resize((20, 20)))
    #         req_img = ImageTk.PhotoImage(Image.open(f'imgs\\icons\\{35:03d}.png').resize((20, 20)))
    #         time_img = ImageTk.PhotoImage(Image.open(f'imgs\\icons\\011.png').resize((20, 20)))
    #
    #         name_label = Label(self.quest_frame, text=name, compound='top', image=quest_img)
    #         name_label.pack(fill='both', expand=1)
    #         name_label.image = quest_img
    #
    #         points_label = Label(self.quest_frame, text=f'+{points}', compound='left', image=points_img)
    #         points_label.pack(fill='both', expand=1)
    #         points_label.image = points_img
    #
    #         req_label = Label(self.quest_frame, text=f'0 / {req}', compound='left', image=req_img)
    #         req_label.pack(fill='both', expand=1)
    #         req_label.image = req_img
    #
    #         time_label = Label(self.quest_frame, text=time, compound='left', image=time_img)
    #         time_label.pack(fill='both', expand=1)
    #         time_label.image = time_img
    #
    #     elif timer is not None:
    #         [w.destroy() for w in self.quest_frame.pack_slaves()]
    #         timer_label = Label(self.quest_frame, text=timer, bg='light cyan')
    #         timer_label.pack(fill='both', expand=1)
    #     else:
    #         pass


class ChooseGUI(tk.Toplevel):
    def __init__(self, parent):
        self.parent = parent
        super(ChooseGUI, self).__init__(parent)

        self.listbox = tk.Listbox(self)
        self.listbox.pack()

        self.populate_listbox()

    def populate_listbox(self):
        with open('test.txt', 'r') as f:
            while True:
                print(f.readline(100))


if __name__ == "__main__":
    GUI().mainloop()
